src/main/java/com/minecolonies/coremod/entity/pathfinding/Pathfinding.java

```diff
@@ -91,153 +90,156 @@ public static void debugDraw(final double frame)
 
         //  Y-
-        GL11.glVertex3d(0.0, 0.0, 1.0);
-        GL11.glVertex3d(0.0, 0.0, 0.0);
-        GL11.glVertex3d(1.0, 0.0, 0.0);
-        GL11.glVertex3d(1.0, 0.0, 1.0);
+        vertexBuffer.pos(0.0, 0.0, 1.0).endVertex();
+        vertexBuffer.pos(0.0, 0.0, 0.0).endVertex();
+        vertexBuffer.pos(1.0, 0.0, 0.0).endVertex();
+        vertexBuffer.pos(1.0, 0.0, 1.0).endVertex();
 
-        GL11.glEnd();
+        tessellator.draw();
 
         if (n.parent != null)
         {
-            GL11.glBegin(GL11.GL_LINES);
-            GL11.glColor3f(0.75F, 0.75F, 0.75F);
-
             final double pdx = n.parent.pos.getX() - n.pos.getX() + 0.125;
             final double pdy = n.parent.pos.getY() - n.pos.getY() + 0.125;
             final double pdz = n.parent.pos.getZ() - n.pos.getZ() + 0.125;
 
-            GL11.glVertex3d(0.5, 0.5, 0.5);
-            GL11.glVertex3d(pdx / 0.25, pdy / 0.25, pdz / 0.25);
-
-            GL11.glEnd();
+            vertexBuffer.begin(GL11.GL_LINES, DefaultVertexFormats.POSITION_COLOR);
+            vertexBuffer.pos(0.5D, 0.5D, 0.5D).color(0.75F, 0.75F, 0.75F, 1.0F).endVertex();
+            vertexBuffer.pos(pdx / 0.25, pdy / 0.25, pdz / 0.25).color(0.75F, 0.75F, 0.75F, 1.0F).endVertex();
+            tessellator.draw();
         }
 
-        GL11.glPopMatrix();
+        GlStateManager.popMatrix();
     }
 
     @SideOnly(Side.CLIENT)
-    private static void renderDebugText(@NotNull final Node n, final float f1)
+    private static void renderDebugText(@NotNull final Node n)
     {
         final String s1 = String.format("F: %.3f [%d]", n.getCost(), n.getCounterAdded());
         final String s2 = String.format("G: %.3f [%d]", n.getScore(), n.getCounterVisited());
         final FontRenderer fontrenderer = Minecraft.getMinecraft().fontRendererObj;
-        GL11.glPushAttrib(GL11.GL_ALL_ATTRIB_BITS);
-        GL11.glPushMatrix();
-        GL11.glTranslatef(0.0F, 0.75F, 0.0F);
-        GL11.glNormal3f(0.0F, 1.0F, 0.0F);
+        GlStateManager.pushAttrib();
+        GlStateManager.pushMatrix();
+        GlStateManager.translate(0.0F, 0.75F, 0.0F);
+        GlStateManager.glNormal3f(0.0F, 1.0F, 0.0F);
 
         final RenderManager renderManager = Minecraft.getMinecraft().getRenderManager();
-        GL11.glRotatef(-renderManager.playerViewY, 0.0F, 1.0F, 0.0F);
-        GL11.glRotatef(renderManager.playerViewX, 1.0F, 0.0F, 0.0F);
-        GL11.glScalef(-f1, -f1, f1);
-        GL11.glTranslatef(0.0F, (float) (0.25D / f1), 0.0F);
-        GL11.glDepthMask(false);
-
-        final Tessellator tessellator = Tessellator.getInstance();
-        final VertexBuffer VertexBuffer = tessellator.getBuffer();
-        GL11.glDisable(GL11.GL_TEXTURE_2D);
+        GlStateManager.rotate(-renderManager.playerViewY, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate(renderManager.playerViewX, 1.0F, 0.0F, 0.0F);
+        GlStateManager.scale(-0.014F, -0.014F, 0.014F);
+        GlStateManager.translate(0.0F, 18F, 0.0F);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 18F to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)